testsuite/MDAnalysisTests/analysis/test_contacts.py

```diff
@@ -181,6 +184,51 @@ def _run_Contacts(self, universe, start=None, stop=None, step=None, **kwargs):
             radius=6.0,
             **kwargs).run(start=start, stop=stop, step=step)
 
+    @pytest.mark.parametrize("seltxt", [sel_acidic, sel_basic])
+    def test_select_valid_types(self, universe, seltxt):
+        """Test if Contacts._get_atomgroup() can take both string and AtomGroup
+         as selections.
+        """
+        ag = universe.select_atoms(seltxt)
+
+        ag_from_string = contacts.Contacts._get_atomgroup(universe, seltxt)
+        ag_from_ag = contacts.Contacts._get_atomgroup(universe, ag)
+
+        assert_equal(ag_from_string, ag_from_ag)
+
+    def test_contacts_selections(self, universe):
+        """Test if Contacts can take both string and AtomGroup as selections.
+        """
+        aga = universe.select_atoms(self.sel_acidic)
+        agb = universe.select_atoms(self.sel_basic)
+
+        cag = contacts.Contacts(
+            universe,
+            select=(aga, agb),
+            refgroup=(aga, agb),
+        ).run()
```

Comment:
It's very pedantic but I'm going to object to these a little bit, this is not very readable.

If you're going to call run on it in one line then please re-arrange so that it's easier to read.

Otherwise close the brackets at 208 and call `cag.run()` separately.

Same thing below please.

Reply:
Opted for rearrangement. Is this better?

Reply:
Sorry my readability issue wasn't with the way the arguments were arranged but instead with the `).run()` being on a single line. The compactness advantage of calling `run()` at the same time as class construction just disappears in this case.

Reply:
I think I get you now. I put `.run()` on a new line altogether in [this latest commit](https://github.com/MDAnalysis/mdanalysis/pull/3565/commits/55636bf304a590b0c6e52dfee76a13e0d4861a48#diff-c79b8e25558e1ba2f042d25eaa0e355502afba57cf03bf8c9b22e0c6103063a4R215).